invoice/invoice.py

```diff
@@ -45,3 +47,36 @@ def get_sorted_key(self, name):
                 key.append(purchase)
 
         return key
+
+
+class InvoiceReport(HTMLReport):
+    __name__ = 'account.invoice'
+
+    @classmethod
+    def __setup__(cls):
+        super(InvoiceReport, cls).__setup__()
+        cls.__rpc__['execute'] = RPC(False)
+
+    @classmethod
+    def execute(cls, ids, data):
+        pool = Pool()
+        Invoice = pool.get('account.invoice')
+
+        result = super(InvoiceReport, cls).execute(ids, data)
+
+        if len(ids) == 1:
+            # Re-instantiate because records are TranslateModel
+            invoice, = Invoice.browse(ids)
+            if invoice.invoice_report_cache:
```

Comment:
Cache should be checked before calling super() because if there's already a cache we do not wan to render the report again, which is what super will do.